app/controllers/user_sessions_controller.rb

```diff
@@ -26,8 +26,14 @@ def handle_social_login_flow(auth)
     unless params[:hash_params].to_s.empty?
       hash_params = URI.parse("#" + params[:hash_params]).to_s
     end
-
-    if signed_in?
+    @user = User.where(email: auth["info"]["email"]) ? User.find_by(email: auth["info"]["email"]) : @identity.user
```

Comment:
changes should go inside `if not signed in`.

Reply:
Why?

Reply:
Do you know about time complexity?
If person is already signed in then there is no need of your logic. 
It should be time optimized. Banned OR moderator user is a rare case. Think in terms of million qps, then this will be a huge overhead. What do you think?

Reply:
Like this?
```ruby
if signed_in?
   # code that is already there for that case
else
   # changes
```

Reply:
Yes

Reply:
Oh yeah ok I understand

Reply:
Did you understand my view point?
It is important from scalability views

Reply:
Does the last commit fix it?
Yeah I totally understand that it's quicker to do it that way